src/systemctl/systemctl-list-units.c

```diff
@@ -44,10 +44,9 @@ static int get_unit_list_recursive(
                 return c;
 
         r = set_put(replies, reply);
-        if (r < 0) {
-                sd_bus_message_unref(reply);
+        if (r < 0)
                 return log_oom();
-        }
+        TAKE_PTR(reply);
```

Comment:
I wanted to say: "use `set_consume()`", but I see that it uses `free`, and not `set->hash_ops->free_value`. I think we'll want to change that, but it's subject for another PR.

Reply:
Will do soon after merge.

Reply:
#26535.